android/app/src/main/java/com/emergetools/hackernews/features/settings/SettingsScreen.kt

```diff
@@ -171,6 +186,52 @@ fun SettingsScreen(
   }
 }
 
+@Composable
+private fun VersionCard() {
+  val context = LocalContext.current
+  val scope = rememberCoroutineScope()
+  var isLoading by remember { mutableStateOf(false) }
+  var status by remember { mutableStateOf<UpdateStatus?>(null) }
+  return SettingsCard(
+    leadingIcon = {
+      Icon(
+        modifier = Modifier.width(12.dp),
+        painter = painterResource(R.drawable.ic_settings),
+        contentDescription = null // Purely decorative
+      )
+    },
+    trailingIcon = {
+      Icon(
+        modifier = Modifier.width(12.dp),
+        painter = painterResource(R.drawable.ic_arrow_up_right),
+        tint = MaterialTheme.colorScheme.onSurface,
+        contentDescription = "Link"
+      )
+    },
+    label = when (isLoading) {
+      true -> "Loading..."
+      else -> when (status) {
+        is UpdateStatus.UpToDate -> "Up to date!"
+        is UpdateStatus.Error -> (status as UpdateStatus.Error).message
+        is UpdateStatus.NewRelease -> "New release!"
+        else -> "Check for updates"
+      }
+    }
+  ) {
+    scope.launch {
+      isLoading = true
+      status = null
+      status = Distribution.checkForUpdate(context)
```

Comment:
How does this work for prod release builds given the intent of distro is for internal releases?

Reply:
I added `Distribution.isEnabled` call recently which we can use to hide the whole section - but I think the larger question is how we want to set up the variants (e.g. should there be a separate `playstoreRelease` variant which has build distro disabled and a `beta` variant which is like release but has this enabled).

Long term it would also be nice to have a `com.emergetools.distribution:distribution-stubs` dependacy in the vein of https://fbflipper.com/docs/getting-started/android-native/#dependencies so you don't even have to include the code in the release app - but so far it's not that much code so it's not a huge deal either way.